### PR TITLE
feat(demo): handle org initialization

### PR DESCRIPTION
**Pull request by scefali**
This PR makes a couple of changes with regards to initializing orgs in demo mode:
1. Adds an explicit initialization status on the `DemoOrganization` model. This is needed because populating the org with actual events happens outside of the transaction. While orgs are being initialized, they are not ready to be assigned to users.
2. If data population fails, delete the organization so we don't have orphaned orgs.
3. Move `ORG_BUFFER_SIZE` to a setting that we can modify without changing code

### CODE REVIEW

File: src/sentry/demo/tasks.py [hunk 1]
```diff
@@ -46,26 +46,22 @@ def delete_users_orgs(**kwargs):
...
 
     # synchronnously build up our org buffer if under sized
     if num_to_populate > 0:
         create_demo_org()
         build_up_org_buffer.apply_async()
-
-
-# on initialization, start building up our org buffer
-if settings.DEMO_MODE:
-    build_up_org_buffer.apply_async()
```
Comment (scefali) [inline src/sentry/demo/tasks.py:71]: this got left by accident, we are kicking off the task when the app loads in the `src/sentry/demo/apps.py` file

**Review by wedamija — APPROVED** [review]
Migration should be fine, only runs on demo and is just changing the default.

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -2160,6 +2160,7 @@ def create_partitioned_queues(name):
     "DEFAULT_BACKOFF_TIME": 0,  # backoff time between sending events
     "ERROR_BACKOFF_TIME": 0.5,  # backoff time after a snuba error
     "NUM_RELEASES": 3,
+    "ORG_BUFFER_SIZE": 3,  # number of pre-populated organizations in the buffer
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/conf/server.py:2163]: @wedamija good point, let me look into moving them

File: src/sentry/demo/demo_org_manager.py [hunk 3]
```diff
@@ -54,12 +57,27 @@ def create_demo_org(quick=False) -> Organization:
...
     logger.info(
-        "create_demo_org.post-transaction", extra={"organization_slug": org.slug, "quick": quick}
+        "create_demo_org.post-transaction",
+        extra={"organization_slug": org.slug, "quick": quick},
     )
-    generate_releases([react_project, python_project], quick=quick)
-    populate_connected_event_scenario_1(react_project, python_project, quick=quick)
+    try:
+        generate_releases([react_project, python_project], quick=quick)
+        populate_connected_event_scenario_1(react_project, python_project, quick=quick)
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/demo/demo_org_manager.py:66]: @wedamija as I add more and more cases, I'm looking to increasing the configurability. Though it's not super straightfoward.

**Pull request merged**